bookserver/routers/rslogging.py

```diff
@@ -50,7 +71,7 @@ async def log_book_event(entry: LogItemIncoming, request: Request):
     if request.state.user:
         entry.sid = request.state.user.username
     else:
-        entry.sid = "Anonymous"
+        return JSONResponse(content={"result": "ignored"})
```

Comment:
I think a function like make_json_response is fine.  I'm assuming it would return the JSONResponse object.

Question: where should we put utilities like this?  __init__.py ??  internal/utils.py ??

Looking at the official status codes:

200
201 - Created : would be good for success when we are creating something

401 - Unauthorized : this seems like the closest response for when an API is called and the user should be logged in but is not. I think this covers all the cases where the detail is ignored right now.  4xx errors are all classified as "client errors" so this makes sense to me.

FAILURE - This goes back to your earlier question... Should the CRUD functions trap errors or should we let them generate 500 errors and return them from middleware?  In any case a 500 seems the most appropriate of the   5xx errors are all classified as server errors so this makes sense also.

So in any case if we use the middleware approach we would still call this function with a FAILURE status, and the detail would likely be either the stack trace provided by the caught exception, or if it is a failure that we choose to catch in some other context the detail that we would provide.

Reply:
I updated to use the make_json_respons, but I like using the status.HTTP_XXX constants provided as it gives us better flexibility that just 3 possible responses.

Reply:
I'll comment more on this, but I'm busy for several more hours...

Reply:
Alright, please take a look at what I've done in rslogging as an example of how I think we can use this and provide a consistent API.  I'm in video meeting all day today but can multitask on this...

Reply:
I agree re: status codes. The design should be DRY and based on KISS -- using status codes instead of duplicating them in JSON is much better. 

I think this function and the related classes belongs in `schema.py`, since that's what helps define the client/server interface.

I agree re: 2xx codes (200, 201, etc.) == success, 4xx codes == not logged in. I can see two flavors of database failures: expected (duplicate username, etc.) that should be reported back to the user with an explanation (this user already exists) vs. unexpected (can't connect to database, etc.) report to the user as "sorry, please try again later". I also agree on sending a stack trace for these unexpected failures (plus logging it as well).